### PR TITLE
Add software / hardware isolation use case

### PR DESCRIPTION
**Pull request by OR13**
- remove the "ephemeral only" use case assertion
- add language regarding software / hardware isolation use case

***
<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/did-method-key/pull/53.html" title="Last updated on Jun 21, 2022, 7:57 PM UTC (5732aa0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/did-method-key/53/a8b5d22...5732aa0.html" title="Last updated on Jun 21, 2022, 7:57 PM UTC (5732aa0)">Diff</a>

### CODE REVIEW

**Review by msporny — APPROVED** [review]
Thank you, this text has been biting us for years. +1 to merge this PR, but I'd like to go further in future text to note that it's perfectly fine to use a did:key in a long-lived way if you have hardware isolation around the private key material and if the key material is not extractable into a non-protected form (extracting to clear text is not ok, but transferring it (encrypted) from one HSM to another is fine).

**Comment by OR13** [comment]
@msporny thank you for the massive update on the did document generation.

> In other words, we probably want to say that "did:key" for long-lived use cases is totally fine as long as you have (ideally, provable) hardware isolation.

I can add a note or advisement on this if you like.

File: index.html [hunk 1]
```diff
@@ -165,8 +165,9 @@ <h2>Introduction</h2>
 guarantees, the general approaches tend to be expensive to setup and operate.
 Some use cases requiring DIDs do not need the guarantees provided by these
 heavy-weight systems. For example, a DID that will only be used for a single,
-ephemeral interaction might not need to be registered, updated, or deactivated.
-It is for this class of use cases that the <tt>did:key</tt> method exists.
+ephemeral interaction might not need to be registered, updated, or deactivated, 
+or where private keys are protected by software or hardware isolation.
+These are a few of the cases where using <tt>did:key</tt> might offer unique benefits.
       </p>
```
Comment (OR13) [inline index.html:172]:
```suggestion
<p class="advisement">
Use of <tt>did:key</tt> for long-lived use cases is only recommended when accompanied with high confidence in hardware isolation.
</p>
```

**Review by dmitrizagidulin — APPROVED** [review]
:+1: 
Thanks Orie!

**Pull request merged**